rust/lancedb/src/remote/table.rs

```diff
@@ -1035,11 +1083,28 @@ impl<S: HttpSend> BaseTable for RemoteTable<S> {
 
         let (request_id, response) = self.send_streaming(request, new_data, true).await?;
 
-        // TODO: server can response with these stats in response body.
-        // We should test that we can handle both empty response from old server
-        // and response with stats from new server.
-        self.check_table_response(&request_id, response).await?;
-        Ok(MergeStats::default())
+        let response = self.check_table_response(&request_id, response).await?;
+        let body = response.text().await.err_to_http(request_id.clone())?;
+
+        if body.trim().is_empty() || body == "{}" {
+            // Backward compatible with old servers
+            let version = self.version().await?;
+            return Ok(MergeResult {
+                version,
+                num_deleted_rows: 0,
+                num_inserted_rows: 0,
+                num_updated_rows: 0,
```

Comment:
This is an unfortunate default, but I think it's okay since this code path will be unused as soon as we update the servers.